architecture/adapters.md

```diff
@@ -12,16 +12,35 @@ They can be written in any language, and must simply respond to a set of HTTP en
 ## Basic Concept
 Here's a diagram illustrating the basic concept of Yacs and adapters:
 
-<!-- ![alt-text](../_media/adaptersfig1.png) -->
+<!-- ![alt-text](../_media/adaptersfig1.png) - ->
 ========  
 DIAGRAM GOES HERE  
 ========  
-<!-- TODO: Make diagram -->
-<!--
-When Yacs needs information that isn't stored in its database, it sends an HTTP request to the adapter that is responsible for that information. Then that adapter works its magic, and sends the information back in JSON format. This document is going to focus on the [`yaml-rpi`][yaml-rpi-adapter] adapter and use its code in examples.
+<!-- TODO: Make diagram --><!--
+
+- ->
+When Yacs needs information that isn't stored in its database, it sends an HTTP request to the adapter that is responsible for that information. Then that adapter works its magic, and sends the information back in JSON format. This document is going to focus on the [`yaml-rpi`][yaml-rpi-adapter] adapter and use its code in examples. The document `app.rb` will also be edited for clarity to this:
+
+``` ruby
+require 'sinatra'
+require 'sinatra/json'
+require 'oj'
+require 'yaml'
+
+set :bind, '0.0.0.0'
+set :port, 4600
+
+ENV['YAML_SOURCE'] ||= 'schools-and-subjects.yml'
+
+get "/:term_shortname" do
+	file = open(ENV['YAML_SOURCE'])
+	yaml_content = YAML.load(file)
+	json yaml_content
+end
+```
 
 ## Data Source
-The first thing your adapter needs is a place to get data from. In the `yaml-rpi` adapter, that place is a file called `schools-and-subjects.yml`.
+The first thing your adapter needs is a place to get data from. In the `yaml-rpi` adapter, that place is a file called `schools-and-subjects.yml`:
```

Comment:
I don't know if this is the best example to start with. It is the simplest, but it is also the least useful. It is not typical that data will be coming from a config file, and I don't necessarily want to encourage people to use adapters that way. Although they can be used like this, they are intended to ingest dynamic data from some other source on the web.

Reply:
Hmmm, do you think this markdown should use a script in another language? There's quite a few ruby/sinatra specific parts for the adapters as far as I can tell. Maybe it'd be better to use python or pseudo-code in the toy example code?